src/main.rs

```diff
@@ -244,12 +248,22 @@ fn remote_server_options(matches: &ArgMatches) -> Result<(Option<Sources>, Optio
                 docker_config_json_path,
             ))?)
         } else {
-            let docker_config_json_path = DEFAULT_ROOT.config_dir().join("config.json");
-            if Path::exists(&docker_config_json_path) {
-                Some(read_docker_config_json_file(&docker_config_json_path)?)
-            } else {
-                None
+            // try reading kubewarden's config or docker config (in this order)
+            // the first configuration file found is going to be returned.
+            let mut configuration_files = vec![DEFAULT_ROOT.config_dir().join("config.json")];
+            if let Some(user_dir) = UserDirs::new() {
+                let config_json_path = user_dir.home_dir().join(".docker").join("config.json");
+                configuration_files.push(config_json_path);
+            }
+
+            let mut conf: Option<DockerConfig> = None;
+            for path in configuration_files.iter() {
+                if Path::exists(&path) {
+                    conf = Some(read_docker_config_json_file(&path)?);
```

Comment:
What happens if one configuration is not compatible, but a subsequent one is? We would bubble up the exception on the first one, right?